doc/js_tutorials/js_core/js_basic_ops/js_basic_ops.markdown

```diff
@@ -0,0 +1,293 @@
+@endcode
+
+**2. at**
+
+@code{.js}
+let x = 3, y = 4;
+let src = cv.imread("canvasInput");
+let pixel = src.ucharAt(y, x * src.channels());
+@endcode
+
+**3. ptr**
+
+@code{.js}
+let x = 3, y = 4;
+let src = cv.imread("canvasInput");
+let pixel = src.ucharPtr(y, x)[0];
+@endcode
+
+Image ROI
+---------
+
+Sometimes, you will have to play with certain region of images. For eye detection in images, first
+face detection is done all over the image and when face is obtained, we select the face region alone
+and search for eyes inside it instead of searching whole image. It improves accuracy (because eyes
+are always on faces) and performance (because we search for a small area)
+
+Try it
+------
+
+Here is a demo. Canvas elements named roiCanvasInput and roiCanvasOutput have been prepared. Choose an image and 
+click `Try it` to see the result. And you can change the code in the textbox to investigate more.
+
+\htmlonly
+<!DOCTYPE html>
+<head>
+<style>
+canvas {
+    border: 1px solid black;
+}
+.err {
+    color: red;
+}
+</style>
+</head>
+<body>
+<div id="roiCodeArea">
+<h2>Input your code</h2>
+<button id="roiTryIt" disabled="true" onclick="roiExecuteCode()">Try it</button><br>
+<textarea rows="7" cols="80" id="roiTestCode" spellcheck="false">
+let src = cv.imread("roiCanvasInput");
+let dst = new cv.Mat();
+// You can try more different parameters
+let rect = new cv.Rect(100, 100, 200, 200);
+dst = src.getRoiRect(rect); 
+cv.imshow("roiCanvasOutput", dst);
+src.delete(); dst.delete(); 
+</textarea>
+<p class="err" id="roiErr"></p>
+</div>
+<div id="roiShowcase">
+    <div>
+        <canvas id="roiCanvasInput"></canvas>
+        <canvas id="roiCanvasOutput"></canvas>
+    </div>
+    <input type="file" id="roiInput" name="file" />
+</div>
+<script src="utils.js"></script>
+<script async src="opencv.js" id="opencvjs"></script>
+<script>
+function roiExecuteCode() {
+    let roiText = document.getElementById("roiTestCode").value;
+    try {
+        eval(roiText);
+        document.getElementById("roiErr").innerHTML = " ";
+    } catch(err) {
+        document.getElementById("roiErr").innerHTML = err;
+    }
+}
+
+loadImageToCanvas("lena.jpg", "roiCanvasInput");
+let roiInputElement = document.getElementById("roiInput");
+roiInputElement.addEventListener("change", roiHandleFiles, false);
+function roiHandleFiles(e) {
+    let roiUrl = URL.createObjectURL(e.target.files[0]);
+    loadImageToCanvas(roiUrl, "roiCanvasInput");
+}
+</script>
+</body>
+\endhtmlonly
+
+
+Splitting and Merging Image Channels
+------------------------------------
+
+Sometimes you will need to work separately on R,G,B channels of image. Then you need to split the
+RGB images to single planes. Or another time, you may need to join these individual channels to RGB
+image. 
+
+@code{.js}
+let src = cv.imread("canvasInput");
+let rgbPlanes = new cv.MatVector();
+cv.split(src, rgbPlanes);
+cv.merge(rgbPlanes, src);
+src.delete(); rgbPlanes.delete();
```

Comment:
And it's better to show how to use mat in vector `rgbPlanes.get`, then you warn developer that they need to delete mat return from `rgbPlanes.get`.